### PR TITLE
Get it building again under 1.0.0-beta (rust-lang/rust@9854143)

### PR DESCRIPTION
**Pull request by robn**

### CODE REVIEW

**Comment by metajack** [comment]
Does this need to go upstream? Or is this a backport of upstream?

**Comment by metajack** [comment]
Disregard my comment. I thought this was glutin.

File: build.rs [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+use std::path::Path;
+use std::fs::File;
+use std::io::BufWriter;
 
 fn main() {
-    let dest = Path::new(os::getenv("OUT_DIR").unwrap());
+    let dest = env::var("OUT_DIR").unwrap();
 
-    let mut file = File::create(&dest.join("gl_bindings.rs")).unwrap();
+    let mut file = BufWriter::new(File::create(&Path::new(&dest).join("gl_bindings.rs")).unwrap());
```
Comment (Ms2ger) [inline build.rs]: Passing &dest to File::create should work too.

File: src/gl.rs [hunk 2]
```diff
@@ -206,7 +205,7 @@ pub fn vertex_attrib_pointer_f32(index: GLuint,
                                 ffi::FLOAT,
                                 normalized as GLboolean,
                                 stride,
-                                mem::transmute(offset as uint));
+                                mem::transmute(offset as *const GLvoid));
```
Comment (Ms2ger) [inline src/gl.rs]: transmute should no longer be necessary.

File: src/gl.rs [hunk 3]
```diff
@@ -307,27 +306,27 @@ pub fn uniform_1i(location: GLint, x: GLint) {
...
+        let mut result: Vec<_> = repeat(0u8).take(1024).collect();
         let mut result_len: GLsizei = 0 as GLsizei;
         ffi::GetProgramInfoLog(program,
                             1024 as GLsizei,
                             &mut result_len,
                             result.as_ptr() as *mut GLchar);
-        result.truncate(if result_len > 0 {result_len as uint - 1u} else {0u});
-        from_utf8(result.as_slice()).unwrap().to_string()
+        result.truncate(if result_len > 0 {result_len as usize - 1} else {0});
+        from_utf8(&*result.into_boxed_slice()).unwrap().to_string()
```
Comment (Ms2ger) [inline src/gl.rs]: Use String::from_utf8 instead.

File: src/gl.rs [hunk 4]
```diff
@@ -358,8 +357,7 @@ pub fn get_string(which: GLenum) -> String {
     unsafe {
         let llstr = ffi::GetString(which);
         if !llstr.is_null() {
-            return str::from_utf8_unchecked(c_str_to_bytes_with_nul(&(llstr as *const c_char))
-                                             ).to_string();
+            return str::from_utf8_unchecked(CStr::from_ptr(llstr as *const i8).to_bytes()).to_string();
```
Comment (Ms2ger) [inline src/gl.rs]: `c_char` isn't `i8` everywhere.

File: build.rs (hunk 1 above)
Comment (robn) [inline build.rs]: Not sure what you mean?

File: src/gl.rs (hunk 2 above)
Comment (robn) [inline src/gl.rs]: Confirmed.

File: src/gl.rs (hunk 3 above)
Comment (robn) [inline src/gl.rs]: :+1:

File: src/gl.rs (hunk 4 above)
Comment (robn) [inline src/gl.rs]: Agg. Fixed.

File: build.rs (hunk 1 above)
Comment (jdm) [inline build.rs]: File::create(&dest)

Comment (robn) [inline build.rs]: &dest is just OUT_DIR at that point. We still need to append gl_bindings.rs.

Comment (Ms2ger) [inline build.rs]: You are correct. Nevermind.

File: build.rs [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-
 extern crate gl_generator;
 extern crate khronos_api;
 
-use std::os;
-use std::old_io::File;
+use std::env;
+use std::path::Path;
+use std::fs::File;
+use std::io::BufWriter;
```
Comment (Ms2ger) [inline build.rs]: Nit: sort those

**Comment by Ms2ger** [comment]
One final question: what's up with the explicit dereferencing around CString::new?

File: build.rs (hunk 5 above)
Comment (robn) [inline build.rs]: Very nit, but ok :)

**Comment by robn** [comment]
Just a lack of understanding, still fairly new to Rust. I've tidied those up.

**Comment by Ms2ger** [comment]
Thanks for bearing with us!

**Comment by robn** [comment]
No better way to learn, thanks! :)

**Pull request merged**